### ISSUE TITLE
UnusedLocalVariable / foreach / double quotes compact bug in 2.1.1

### ISSUE THREAD

**Issue by demiankatz** [phpmd/phpmd#194]
I'm using PHPMD 2.1.1, and while this solves several compact()-related problems found in earlier releases, there still appears to be an issue where foreach and compact trigger a false UnusedLocalVariable warning when variable names are enclosed in double quotes. Example:

```` php
<?php
class Example
{
    protected function f($a)
    {
        foreach ($a as $k => $v) {
            return compact("k", "v");
        }
    }
}
````

If you replace the compact line with:

``return compact('k', 'v');``

(substituting single for double quotes)

then the warnings go away. Hopefully that means this is a fairly straightforward problem to fix!

**Comment by ravage84** [comment]
Related to https://github.com/phpmd/phpmd/issues/224
